doc/classes/RenderingDevice.xml

```diff
@@ -1583,22 +1583,22 @@
 		<constant name="BARRIER_MASK_VERTEX" value="1" enum="BarrierMask" is_bitfield="true">
 			Vertex shader barrier mask.
 		</constant>
-		<constant name="BARRIER_MASK_FRAGMENT" value="2" enum="BarrierMask" is_bitfield="true">
+		<constant name="BARRIER_MASK_FRAGMENT" value="8" enum="BarrierMask" is_bitfield="true">
 			Fragment shader barrier mask.
 		</constant>
-		<constant name="BARRIER_MASK_COMPUTE" value="4" enum="BarrierMask" is_bitfield="true">
+		<constant name="BARRIER_MASK_COMPUTE" value="2" enum="BarrierMask" is_bitfield="true">
 			Compute barrier mask.
 		</constant>
-		<constant name="BARRIER_MASK_TRANSFER" value="8" enum="BarrierMask" is_bitfield="true">
+		<constant name="BARRIER_MASK_TRANSFER" value="4" enum="BarrierMask" is_bitfield="true">
 			Transfer barrier mask.
 		</constant>
-		<constant name="BARRIER_MASK_RASTER" value="3" enum="BarrierMask" is_bitfield="true">
+		<constant name="BARRIER_MASK_RASTER" value="9" enum="BarrierMask" is_bitfield="true">
```

Comment:
Old code that used the `BARRIER_MASK_RASTER` flag will be equivalent to using only `BARRIER_MASK_VERTEX` now. Is that OK? Or is the `BARRIER_MASK_FRAGMENT` also important?

Reply:
Remember we're comparing to what is currently released. So this is a change from "0" to "9", but yes the conclusion is the same. If you used `BARRIER_MASK_RASTER` in an extension in 4.1 and you now run this same extension in 4.2 without recompiling, you will only get a barrier with the vertex shader.

Seeing you can't really run a fragment shader without first running the vertex shader, this is likely safe. 

The original enhancement we're after here is when the vertex shader should not wait for execution before a previous process is finished which was not possible in the past.
As on mobile the vertex shader for all geometry in a pass is run fully before before the fragment shader is applied, this will allow you to run data transfers (populating UBOs) and compute shaders in parallel with the vertex shader. 
On desktop this is a non-issue as vertex and fragment shaders run in parallel.

Reply:
Just barrier mask vertex is fine!